internal/clusterfeature/features/securityscan/spec.go

```diff
@@ -0,0 +1,119 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package securityscan
+
+import (
+	"emperror.dev/errors"
+	"github.com/banzaicloud/pipeline/internal/clusterfeature"
+	"github.com/mitchellh/mapstructure"
+)
+
+//featureSpec security scan cluster feature specific specification
+type featureSpec struct {
+	CustomAnchore    anchoreSpec       `json:"customAnchore" mapstructure:"customAnchore"`
+	Policy           policySpec        `json:"policy" mapstructure:"policy"`
+	ReleaseWhiteList []releaseSpec     `json:"releaseWhiteList,omitempty" mapstructure:"releaseWhiteList"`
+	WebhookConfig    webHookConfigSpec `json:"webhookConfig" mapstructure:"webhookConfig"`
+}
+
+// Validate validates the input security scan specification.
+func (s featureSpec) Validate() error {
+
+	var validationErrors error
+
+	if s.CustomAnchore.Enabled {
+		validationErrors = s.CustomAnchore.Validate()
+	}
+
+	if !s.Policy.CustomPolicy.Enabled && s.Policy.PolicyID == "" {
+		validationErrors = errors.Combine(validationErrors, errors.New("policyId is required"))
+	}
+
+	for _, releaseSpec := range s.ReleaseWhiteList {
+		validationErrors = errors.Combine(validationErrors, releaseSpec.Validate())
+	}
+
+	validationErrors = errors.Combine(validationErrors, s.WebhookConfig.Validate())
+
+	return validationErrors
+}
+
+type anchoreSpec struct {
+	Enabled  bool   `json:"enabled" mapstructure:"enabled"`
+	Url      string `json:"url" mapstructure:"url"`
+	SecretID string `json:"secretId" mapstructure:"secretId"`
+}
+
+func (a anchoreSpec) Validate() error {
+
+	if a.Enabled {
+		if a.Url == "" && a.SecretID == "" {
```

Comment:
I think that `&&` was meant to be an `||` instead.

Reply:
thx